java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -412,8 +414,7 @@ abstract class AbstractArcHost(
         spec: Plan.Particle,
         particle: Particle,
         scheduler: Scheduler
-    ) = context.particles[spec.particleName]?.copyWith(particle)
-            ?: ParticleContext(particle, spec, scheduler)
+    ) = ParticleContext(particle, spec, scheduler)
```

Comment:
This creates a new ParticleContext everytime. ParticleContext stores per-particle lifecycle state, so this effectively resets everything.

Reply:
I think I need to talk through the lifecycle of this with you. That's actually the bug (in the small) - the second of two particles with the same name ended up in the wrong lifecycle state because it was copyWith'd.